app/javascript/spec/dual-list-select/dual-list-select.spec.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import DualList from '../../components/dual-list-select';
+import List from '../../components/dual-list-select/list';
+
+const FieldProvider = ({ component, ...props }) => (
+  <div>{ component({ input: { name: 'Foo', onChange: jest.fn() }, meta: { error: false, touched: false }, ...props }) }</div>
+);
+
+describe('Dual list component', () => {
+  const props = {
+    options: {
+      key1: 'text1',
+      key2: 'text2',
+      key3: 'text3',
+      key4: 'text4',
+    },
+    input: {
+      value: { key3: 'text3' },
+      onChange: jest.fn(),
+    },
+  };
+  let onChangeSpy;
+
+  afterEach(() => {
+    jest.clearAllMocks();
```

Comment:
This is not that safe. I would create a constant for onChange method and assign that to the input props and the just clear mock for that spy:

```javascript
const onChangeSpy = jest.fn();

const props = {
  ...,
  input: {
    ...,
    onChange: onChangeSpy
  }
}

afterEach(() => {
  onChangeSpy.clearMock(); // probably different method name can't remember right now.
}

```